src/libsystemd-network/dhcp-network.c

```diff
@@ -186,16 +186,18 @@ int dhcp_network_bind_udp_socket(int ifindex, be32_t address, uint16_t port, int
                         return r;
         }
 
+        r = setsockopt_int(s, SOL_SOCKET, SO_BROADCAST, true);
```

Comment:
Please add the condition `if (port == DHCP_PORT_SERVER)`.

Reply:
Sure, I thought about this too